lib/src/gauge.rs

```diff
@@ -72,3 +76,66 @@ impl StatisticSetGauge {
         })
     }
 }
+
+/// A GaugeDimensions is a group of dimensions to be used with a gauge.
+/// It is consumed when the gauge is created.
+#[derive(Debug, Default)]
+pub struct GaugeDimensions {
+    pub dimension_position: DimensionPosition,
+}
+
+impl GaugeDimensions {
+    /// Create a new, empty, GaugeDimensions
+    pub fn new() -> Self {
+        Self {
+            dimension_position: Default::default(),
+        }
+    }
```

Comment:
Is there a reason why this type should not have a trimmer api? E.g.,

```rust
/// Create a new GaugeDimensions
/// ```
/// # use goodmetrics::gauge::GaugeDimensions;
/// GaugeDimensions::new([("a", "dimension"), ("another", "dimension")]);
/// ```
pub fn new(dimensions: impl IntoIterator<Item = (impl Into<Name>, impl Into<Dimension>)>) -> Self {
    Self {
        dimension_position: BTreeMap::from_iter(dimensions.into_iter().map(|(name, dimension)| (name.into(), dimension.into()))),
    }
}
```

and no "from" "with". You already have a Default impl derived, so there's no value to this `new()` as-is. Default + insert accounts for people who want to go one-by-one, and new() in this way accounts for people who want to express their dimensions declaratively.

Reply:
That's much better! Thanks.